app/src/main/java/com/infomaniak/drive/data/services/BulkDownloadWorker.kt

```diff
@@ -0,0 +1,286 @@
+import com.infomaniak.drive.extensions.letAll
+import com.infomaniak.drive.utils.AccountUtils
+import com.infomaniak.drive.utils.MediaUtils
+import com.infomaniak.drive.utils.MediaUtils.isMedia
+import com.infomaniak.drive.utils.NotificationUtils.cancelNotification
+import com.infomaniak.drive.utils.NotificationUtils.downloadProgressNotification
+import com.infomaniak.drive.utils.NotificationUtils.notifyCompat
+import com.infomaniak.lib.core.networking.HttpClient
+import com.infomaniak.lib.core.networking.HttpUtils
+import com.infomaniak.lib.core.utils.SentryLog
+import io.sentry.Sentry
+import kotlinx.coroutines.*
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import java.io.BufferedInputStream
+import java.io.File as IOFile
+
+class BulkDownloadWorker(context: Context, workerParams: WorkerParameters) : CoroutineWorker(context, workerParams) {
+
+    private var notificationManagerCompat: NotificationManagerCompat = NotificationManagerCompat.from(applicationContext)
+    private var filesPair: MutableMap<Int, Pair<File?, IOFile?>> = mutableMapOf()
+
+    private val fileIds: IntArray by lazy { inputData.getIntArray(FILE_IDS) ?: intArrayOf() }
+    private val userDrive: UserDrive by lazy {
+        UserDrive(
+            userId = inputData.getInt(USER_ID, AccountUtils.currentUserId),
+            driveId = inputData.getInt(DRIVE_ID, AccountUtils.currentDriveId)
+        )
+    }
+    private val filesCount by lazy { fileIds.size }
+    private val downloadProgressNotification by lazy { createDownloadNotification() }
+
+    private var numberOfFilesDownloaded = 0
+    private var lastUpdateProgressMillis = System.currentTimeMillis()
+
+    override suspend fun doWork(): Result {
+        return runCatching {
+            SentryLog.i(TAG, "Work started")
+            initOfflineDownload()
+        }.getOrElse { exception ->
+            exception.printStackTrace()
+            when (exception) {
+                is CancellationException -> {
+                    clearFiles()
+                    notifyDownloadCancelled()
+                    Result.failure()
+                }
+                is UploadTask.NetworkException -> {
+                    Result.failure()
+                }
+                is RemoteFileException -> {
+                    Sentry.captureException(exception)
+                    Result.failure()
+                }
+                else -> {
+                    SentryLog.e(TAG, "Failure", exception)
+                    Result.failure()
+                }
+            }
+        }.also {
+            notificationManagerCompat.cancel(NOTIFICATION_ID)
+        }
+    }
+
+    private fun clearFiles() {
+        for ((_, value) in filesPair.entries) {
+            Pair(value.first, value.second).letAll { file, offlineFile ->
+                if (offlineFile.exists() && !file.isIntactFile(offlineFile)) offlineFile.delete()
+            }
+        }
+    }
```

Comment:
What do you mean by checking the current ?

Reply:
The current file, in other words, the last file that you tried to download, for those past and future we do not need to check them since you are processing them synchronously